pkg/controller/storageprofile-controller.go

```diff
@@ -101,8 +102,8 @@ func (r *StorageProfileReconciler) reconcileStorageProfile(sc *storagev1.Storage
 
 	if len(storageProfile.Spec.ClaimPropertySets) > 0 {
 		for _, cps := range storageProfile.Spec.ClaimPropertySets {
-			if len(cps.AccessModes) == 0 && cps.VolumeMode != nil {
-				err = fmt.Errorf("must provide access mode for volume mode: %s", *cps.VolumeMode)
+			if cps.VolumeMode == nil || len(cps.AccessModes) == 0 {
```

Comment:
`cps.VolumeMode == nil` is redundant.  If it is nil len(cps.AccessModes) will return 0

Reply:
How is `cps.VolumeMode == nil` related to  `len(cps.AccessModes)`?

Reply:
lol it's not